src/packages/@ncigdc/components/Oncogrid/getQueries.js

```diff
@@ -117,51 +109,49 @@ async function getQueries({
         },
       ],
     },
-    currentFilters,
+    geneFilters,
   );
   const {
     data: { hits: cases, pagination: { total: totalCases } },
   } = await getCases({
-    filters: caseFilters, //note: do not pass sssm.consequence.transcript.is_canonical here, field does not exist under case_centric
+    filters: caseFilters,
     size: maxCases,
   });
   if (!totalCases) return NO_RESULT;
 
   const caseIds = cases.map(c => c.case_id);
-  const occurrenceFilters = addInFilters(
-    replaceFilters(
-      {
-        op: 'and',
-        content: [
-          {
-            op: 'in',
-            content: { field: 'cases.case_id', value: caseIds },
-          },
-        ],
-      },
-      caseFilters,
-    ),
+  const occurrenceFilters = replaceFilters(
     {
       op: 'and',
       content: [
         {
           op: 'in',
-          content: {
-            field: 'ssm.consequence.transcript.is_canonical',
-            value: ['true'],
-          },
+          content: { field: 'cases.case_id', value: caseIds },
         },
-        noMissingImpact,
       ],
     },
+    caseFilters,
   );
   const { data: { hits: occurrences } } = await getOccurrences({
     filters: occurrenceFilters,
   });
 
+  // front end filter is_canonical on REST endpoint instead of
+  // graphql endpoint with inner hits query because
+  // that is very slow for large size (might be because inner hits causes ES to fire secondary queries)
+  const isCanonicalOccurrences = occurrences.map(occurrence => ({
+    ...occurrence,
+    ssm: {
+      ...occurrence.ssm,
+      consequence: ((occurrence.ssm || {}).consequence || []).filter(
```

Comment:
that optional chaining would sure be nice here eh?

Reply:
🙏